.github/workflows/ci-lint-test.yml

```diff
@@ -6,29 +6,80 @@ name: Lint Checks
   group: ${{ github.workflow }}-${{ (github.event.pull_request && github.event.pull_request.number) || github.ref || github.run_id }}
   cancel-in-progress: true
 
-# See https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
-permissions:  # added using https://github.com/step-security/secure-workflows
+permissions:
   contents: read
 
 jobs:
-  generated-files-check:
+  setup-scripts:
     runs-on: ubuntu-latest
     steps:
-      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+      - uses: step-security/harden-runner@v2
         with:
           egress-policy: audit
+      - name: Clone Jaeger main repo scripts
+        run: |
+          git clone --depth 1 https://github.com/jaegertracing/jaeger.git /tmp/jaeger
+          cp -r /tmp/jaeger/scripts ./scripts
 
-      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
+  lint:
+    needs: setup-scripts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7
         with:
-          submodules: recursive
+          egress-policy: audit
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Run lint checks
+        run: |
+          make lint
+
+  dco-check:
+    needs: setup-scripts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7
+        with:
+          egress-policy: audit
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Run DCO check
+        run: python3 scripts/lint/dco_check.py -b main -v --exclude-pattern '@users\.noreply\.github\.com'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  generated-files-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7
+        with:
+          egress-policy: audit
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+        with:
+          submodules: recursive
       - name: Verify Protobuf types are up to date
-        run: make new-proto && git diff --name-status --exit-code
\ No newline at end of file
+        run: make proto && git diff --name-status --exit-code
+      - name: Verify Thrift types are up to date
+        run: make thrift && git diff --name-status --exit-code
+
+  lint-shell-scripts:
+    needs: setup-scripts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7
+        with:
+          egress-policy: audit
+      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+      - run: sudo apt-get install shellcheck
+      - run: shellcheck scripts/**/*.sh
```

Comment:
we don't have scripts in this repo, why do we need to lint what we download from elsewhere?